drodrpg/DROD/MapWidget.cpp

```diff
@@ -928,8 +928,11 @@ bool CMapWidget::SelectRoomIfValid(
 	if (dwRoomID) //Yes.
 	{
 		MapState storedMapState = MapState::Invisible;
-		if (this->pCurrentGame && this->pCurrentGame->pTotalMapStates)
-			storedMapState = this->pCurrentGame->pTotalMapStates->GetStoredMapStateForRoom(dwRoomID);
+		{//scoping
+			CCurrentGame* pGame = const_cast<CCurrentGame*>(this->pCurrentGame);
```

Comment:
Place const_cast inside the inner if block. Shouldn't const cast a null pointer.